packages/@react-spectrum/tag/src/TagGroup.tsx

```diff
@@ -73,24 +85,19 @@ function TagGroup<T extends object>(props: SpectrumTagGroupProps<T>, ref: DOMRef
         // Refs can be null at runtime.
         let currContainerRef: HTMLDivElement | null = containerRef.current;
         let currTagsRef: HTMLDivElement | null = tagsRef.current;
-        if (!currContainerRef || !currTagsRef) {
-          return;
-        }
-
-        let tags = [...currTagsRef.children];
-        let buttons = [...currContainerRef.parentElement.querySelectorAll('button')];
-        if (tags.length === 0 || buttons.length === 0) {
+        let tags = currTagsRef ? [...currTagsRef.children] : [];
+        if (!currContainerRef || tags.length === 0) {
```

Comment:
Can we check `state.collection.size === 0` here instead? The render empty state div is a child of `currTagsRef` so `tags.length` won't ever be 0.